cmd/build.go

```diff
@@ -58,6 +59,8 @@ func init() {
 
 	buildCmd.Flags().StringVar(&golangImage, "golang-image", "quay.io/giantswarm/golang", "golang image")
 	buildCmd.Flags().StringVar(&golangVersion, "golang-version", "1.13.1", "golang version")
+
+	publishCmd.Flags().StringVar(&channels, "channels", "", "channels to publish the charts to, separated by comma")
```

Comment:
How would that work? Would we have now to update all the CI configs and set the channel if the branch is `master`?

Reply:
```suggestion
	publishCmd.Flags().StringVar(&channels, "channels", "stable", "channels to publish the charts to, separated by comma")
```

I think it should default to stable for backwards compatibility. So we only change the circle config if we see this problem.

Reply:
I'm thinking if we can skip flags all together and check `CIRCLE_BRANCH` env var. If it's set and it's set to a different value than `master` then we skip pushing to stable channel. No need for flags at all.

Reply:
@rossf7 To be backwards compatible, the channel shouldn't be passed. We weren't using `stable`, we werent't passing anything.

Reply:
> the channel shouldn't be passed.

@fiunchinho Got it. 👍 

@kopiczko I like your idea to check `CIRCLE_BRANCH`. The Circle CI config should prevent this. But sometimes this doesn't work so this would be a fallback.

https://github.com/giantswarm/apiextensions/blob/5bd95e0af5b6b99187b0009d124bf8a232f920f8/.circleci/config.yml#L17-L19

Reply:
For my understanding: where/when would you like to specify the flag?

Reply:
in the apiextensions repository, we don't want the unmerged PR's to overwrite the stable channel

Reply:
> I'm thinking if we can skip flags all together and check `CIRCLE_BRANCH` env var. If it's set and it's set to a different value than `master` then we skip pushing to stable channel. No need for flags at all.

something like
```
branch := os.Getenv("CIRCLE_BRANCH")
	if branch != "master" {
		return w, nil
	}

	helmTasks, err := processHelmDir(fs, projectInfo, NewHelmPushTask)
	if err != nil {
		return nil, microerror.Mask(err)
	}

	w = append(w, helmTasks...)

	return w, nil
```

Reply:
This is the build config of `apiextensions` https://github.com/giantswarm/apiextensions/blob/e11354aee7703fa74c504978a3d4b525c7814730/.circleci/config.yml#L1-L19

```yaml
version: 2
jobs:
  build:
    machine: true
    steps:
    - checkout

    - run: |
        wget -q $(curl -sS -H "Authorization: token $RELEASE_TOKEN" https://api.github.com/repos/giantswarm/architect/releases/latest | grep browser_download_url | head -n 1 | cut -d '"' -f 4)
        chmod +x ./architect
        ./architect version
    - run: ./architect build
    - store_test_results:
        path: /tmp/results
    - deploy:
        command: |
          if [ "${CIRCLE_BRANCH}" == "master" ]; then
            ./architect deploy
          fi
```

What is pushing to the `stable` channel on non-master builds? `architect build`?

Reply:
yes

Reply:
https://github.com/giantswarm/architect/blob/1017c7b25adeeb2d353e3f576943ae17b080b634/workflow/workflow.go#L188

Reply:
Then your prototype from here https://github.com/giantswarm/architect/pull/359#discussion_r377718731 should be fine with one caveat. When the env var doesn't exist at all we need to preserve the old behaviour. There is non-zero chance someone uses `architect build` for local testing.